src/oc/hocdec.h

```diff
@@ -140,17 +143,85 @@ struct Symbol { /* symbol table entry */
 
 using hoc_List = hoc_Item;
 
-union Datum { /* interpreter stack type */
-    double val;
-    Symbol* sym;
-    int i;
-    double* pval; /* first used with Eion in NEURON */
-    Object** pobj;
-    Object* obj; /* sections keep this to construct a name */
-    char** pstr;
-    hoc_Item* itm;
-    hoc_List* lst;
-    void* _pvoid; /* not used on stack, see nrnoc/point.cpp */
+namespace neuron::detail {
+struct _Datum { /* interpreter stack type */
+    union {
+        double val;
+        Symbol* sym;
+        int i;
+        double* pval; /* first used with Eion in NEURON */
+        Object** pobj;
+        Object* obj; /* sections keep this to construct a name */
+        char** pstr;
+        hoc_Item* itm;
+        hoc_List* lst;
+        void* _pvoid; /* not used on stack, see nrnoc/point.cpp */
+    };
```

Comment:
I suggest we name this member (`m_data`?) to be a bit more explicit than using `this`

Reply:
I find it reads a bit better with it "techincally" being a union wrapper

Reply:
I think we might need to cast `this` to be the base class type before casting in derived class methods in that case.